mobilizon_bots/publishers/telegram.py

```diff
@@ -1,50 +1,111 @@
+            url=f"https://api.telegram.org/bot{attrs['token']}/sendMessage",
+            params={"chat_id": attrs["chat_id"], "text": self.message},
         )
-        return self._validate_response(res)
+        try:
+            self._validate_response(res)
+            return True
+        except InvalidResponse:
+            return False
 
-    def validate_credentials(self) -> bool:
-        chat_id = self.credentials.get("chat_id")
-        token = self.credentials.get("token")
-        username = self.credentials.get("username")
-        if all([chat_id, token, username]):
-            # TODO: add composable errors to highlight which credentials are missing
-            self._log_error_and_raise("Some credentials are missing")
+    def validate_credentials(self):
+        attrs = self.get_attrs_from_conf()
+        chat_id = attrs.get("chat_id")
+        token = attrs.get("token")
+        username = attrs.get("username")
+        err = []
+        if not chat_id:
+            err.append("chat ID")
+        if not token:
+            err.append("token")
+        if not username:
+            err.append("username")
+        if err:
+            self._log_error_and_raise(
+                InvalidCredentials, ", ".join(err) + " is/are missing"
+            )
 
         res = requests.get(f"https://api.telegram.org/bot{token}/getMe")
         data = self._validate_response(res)
 
         if not username == data.get("result", {}).get("username"):
-            self._log_error_and_raise("Found a different bot than the expected one")
-        return data
+            self._log_error_and_raise(
+                InvalidBot, "Found a different bot than the expected one"
+            )
 
-    def validate_event(self) -> bool:
-        text = self.event.get("text")
+    def validate_event(self) -> None:
+        text = self.event.description
         if not (text and text.strip()):
-            self._log_error_and_raise("No text was found. Invalid event")
+            self._log_error_and_raise(InvalidEvent, "No text was found")
 
     def _validate_response(self, res):
         res.raise_for_status()
 
         try:
             data = res.json()
-        except ValueError as e:
-            self._log_error("Server returned invalid json data")
-            raise ValueError from e
+        except Exception as e:
+            self._log_error_and_raise(
+                InvalidResponse, f"Server returned invalid json data: {str(e)}"
+            )
 
         if not data.get("ok"):
-            raise ValueError(f"Invalid request (response: {data})")
+            self._log_error_and_raise(
+                InvalidResponse, f"Invalid request (response: {data})"
+            )
 
         return data
+
+    def get_attrs_from_conf(self, log=None):
+        if log is None:
+            log = self._log_error
+        try:
+            conf = settings.PUBLISHER.telegram
+        except AttributeError:
```

Comment:
tutto questo affare qua non serve più perché la config è già validata a monte.

Reply:
ti direi proprio di levare la funzione e accedere elemento per elemento dove ti serve, senza stare a costruire e passare in giro un dict

Reply:
@chobeat rimedio asap